api/v1alpha1/llamastackdistribution_types.go

```diff
@@ -14,17 +14,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Code generated by controller-gen. DO NOT EDIT.
+// nolint:gci,nolintlint
+
 package v1alpha1
 
+//nolint:gci
 import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // DistributionType defines the distribution configuration for llama-stack.
+// +kubebuilder:validation:XValidation:rule="(self.name == '' && self.image != '') || (self.name != '' && self.image == '')",message="Only one of name or image can be specified"
+// +kubebuilder:validation:XValidation:rule="self.name != '' || self.image != ''",message="Either name or image must be specified"
 type DistributionType struct {
-	// Name is the distribution name that maps to a predefined image in imageMap
+	// Name is the distribution name that maps to supported distributions. Currently supported distributions are ollama and vllm.
```

Comment:
```suggestion
	// Name is the distribution name that maps to supported distributions.
```

So that we don't need to edit this comment every time we had a new distro

Reply:
Yeah makes sense

Reply:
Thinking more about this, we might want to include supported distro mapping in description. See [comment](https://github.com/opendatahub-io/llama-stack-k8s-operator/issues/21#issuecomment-2835612241)

Reply:
Can we do this without updating the line each time a new distro is added?

Reply:
I can leave with the update and having the `explain` is important. Thanks